src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1507,172 @@ public void addUpdateRemoveConditionalFormat() throws Exception
         assertEquals(true, editedAquamap.get("bold"));
     }
 
+
+    @Test
+    public void testExportFields() throws Exception
+    {
+        String listName = "exportFieldsTestList";
+
+        FieldDefinition.LookupInfo lookupInfo = new FieldDefinition.LookupInfo(getProjectName(), "lists", listName);
+        ConditionalFormatFilter eqFive = new ConditionalFormatFilter("5", Filter.Operator.EQUAL);
+        FieldDefinition.RangeValidator soonValidator = new FieldDefinition.RangeValidator("soonValidator", "between now and 2 days from now", "aaaa",
+                FieldDefinition.RangeType.GTE, "2020-11-04",
+                FieldDefinition.RangeType.LTE, "2020-11-06");
+        TestDataGenerator dgen = new TestDataGenerator(lookupInfo)
+                .withColumns(List.of(
+                        new FieldDefinition("textField", FieldDefinition.ColumnType.String)
+                            .setDescription("is texty").setLabel("text field"),
+                        new FieldDefinition("booleanField", FieldDefinition.ColumnType.Boolean)
+                            .setLabel("Boolean Field")
+                            .setRequired(true)
+                            .setURL("fake/list-def.url"),
+                        new FieldDefinition("attachmentField", FieldDefinition.ColumnType.Attachment)
+                            .setHidden(true),
+                        new FieldDefinition("decimalField", FieldDefinition.ColumnType.Decimal).
+                            setScale(1234).setFormat("0.####"),
+                        new FieldDefinition("intField", FieldDefinition.ColumnType.Integer)
+                            .setMvEnabled(true)
+                            .setConditionalFormats(List.of(new ConditionalFormat(List.of(eqFive), true, true, false))),
+                        new FieldDefinition("dateTimeField", FieldDefinition.ColumnType.DateAndTime)
+                            .setValidators(List.of(soonValidator))));
+        dgen.createList(createDefaultConnection(), "Key");
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", listName);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldsPanel();
+
+        File exportFile = domainFormPanel.clickExportFields();
+        List<PropertyDescriptor> createdFields = getFieldsFromExportFile(exportFile);
+
+        PropertyDescriptor stringField = createdFields.stream().filter(a-> a.getName().equals("textField"))
+                .findFirst().orElse(null);
+        assertNotNull("expected string field was not exported", stringField);
+        assertThat("description did not export", stringField.getDescription(), is("is texty"));
+        assertThat("label did not export", stringField.getLabel(), is("text field"));
```

Comment:
If you put the initial FieldDefinitions into variables, you could use them here instead of duplicating the text strings for the label/descriptions/etc.

Reply:
I've taken that code out, it is redundant.  The validate function covers all of these points for all fields now

Reply:
Good call.  I've factored them into a list and am using them to validate